Add docker

EvgeniyaKuzmina opened the pull request.

EvgeniyaKuzmina committed:
fix: Удалены лишние выбросы исключений из методов

EvgeniyaKuzmina committed:
feat: приложение разбито на два подприложения: gateway и server.
gateway отвечает за взаимодействие с пользователем. Запросы из gateway отправляются в server.
server в свою очередь взаимодействует с БД.
В gateway реализована вся проверка входных данных в классах-контроллерах. Добавлены соответствующие исключение и отлов этих исключений, добавлены классы Dto.
В классы-контроллеры server поступают уже чистые данные.

EvgeniyaKuzmina committed:
feat: настроен файл docker-compose.yaml для запуска приложений через docker-compose.

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты
fix: 1. исправлены ошибки в методах контроллерах в gateway (изменены названия у запросных данных, удалены лишние параметры у аннотаций)
2. удалены лишние аннотации в классах контролерах приложения server

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты

EvgeniyaKuzmina committed:
style: удалены лишние импорты
fix: 1. удалена проверка параметров в классах контроллерах приложения server
2. добавлено логирование ошибок в классах контроллерах приложения gateway
3. удалены лишние тесты для контроллеров из приложения server

EvgeniyaKuzmina committed:
fix: добавлена проверка, если пользователь ввёл пустой текст для поиска вещи по названию или описанию

Troitsa commented on server/src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -47,11 +43,6 @@ public class ItemController {
     @PostMapping
     public ItemDto createItem(@Valid @RequestBody ItemDto itemDto, @RequestHeader(HEADER_REQUEST) Long userId)
```
валидация теперь ответственность нового сервиса gateway, из серверной части нужно ее убрать. Проще всего будет, если убрать из pom.xml саму зависимость - сразу же будет видно, где она осталась : 

```
<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-validation</artifactId>
			<version>2.7.3</version>
		</dependency>
```

Troitsa commented on gateway/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -0,0 +1,41 @@
+@Data
+@Builder
+public class ItemDto {
+    Collection<CommentDto> comments;
```
модификатор

EvgeniyaKuzmina committed:
style: 1. изменено наследование для всех исключений, теперь исключения наследуются от RuntimeException
2. удалена валидация из класса контроллера item.
3. Добавлены модификаторы доступа к полям в ItemDto

EvgeniyaKuzmina committed:
style: удалён закомментированный код

EvgeniyaKuzmina committed:
style: добавлена пустая строка между методами

EvgeniyaKuzmina committed:
style: добавлена пустая строка между методами

EvgeniyaKuzmina committed:
style: добавлена пустая строка между методами
